core/vm/eips.go

```diff
@@ -192,3 +194,125 @@ func opPush0(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]by
 	scope.Stack.push(new(uint256.Int))
 	return nil, nil
 }
+
+// enable4200 applies EIP-4200 (RJUMP and RJUMPI opcodes)
+func enable4200(jt *JumpTable) {
+	jt[RJUMP] = &operation{
+		execute:     opRjump,
+		constantGas: GasFastStep,
+		minStack:    minStack(0, 0),
+		maxStack:    maxStack(0, 0),
+		eof1:        true,
+	}
+	jt[RJUMPI] = &operation{
+		execute:     opRjumpi,
+		constantGas: GasFastishStep,
+		minStack:    minStack(1, 1),
+		maxStack:    maxStack(1, 1),
+		eof1:        true,
+	}
+}
+
+// opRjump implements the RJUMP opcode
+func opRjump(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
+	var (
+		code = scope.Contract.Container.CodeAt(scope.ActiveSection)
+		idx  = *pc + 1
+	)
+
+	relativeOffset, err := parseArg(code[idx : idx+2])
+	if err != nil {
+		return nil, err
+	}
+
+	// Move PC past the RJUMP instruction and its immediate argument.
+	*pc += 2 + 1
+
+	// Calculate the new PC given the relative offset. Already validated,
+	// so no need to verify casts.
+	*pc = uint64(int64(*pc)+int64(relativeOffset)) - 1 // pc will also be increased by interpreter loop
+
+	return nil, nil
+}
+
+// opRjumpi implements the RJUMPI opcode
+func opRjumpi(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
+	condition := scope.Stack.pop()
+	if condition.BitLen() == 0 {
+		// Not branching, just skip over immediate argument.
+		*pc += 2
+		return nil, nil
+	}
+	return opRjump(pc, interpreter, scope)
+}
+
+// enable4750 applies EIP-4750 (CALLF and RETF opcodes)
+func enable4750(jt *JumpTable) {
+	jt[JUMP].legacyOnly = true
+	jt[JUMPI].legacyOnly = true
```

Comment:
The way it's phrased currently: https://eips.ethereum.org/EIPS/eip-4750

> This EIP aims to remove the need for dynamic jumps as it offers the most important feature those are used for: calling into and returning from functions. While it removes the need, it **does not disallow those instructions**.

(emphasis mine)

Reply:
But I think we *should* disallow those instructions, otherwise the whole thing makes less sense to me

Reply:
FYI the EIP was updated to clarify this (they are disallowed).